server/service.go

```diff
@@ -259,46 +263,51 @@ func (m *BoostService) handleRegisterValidator(w http.ResponseWriter, req *http.
 // handleGetHeader requests bids from the relays
 func (m *BoostService) handleGetHeader(w http.ResponseWriter, req *http.Request) {
 	vars := mux.Vars(req)
-	slot := vars["slot"]
-	parentHashHex := vars["parent_hash"]
-	pubkey := vars["pubkey"]
-	log := m.log.WithFields(logrus.Fields{
-		"method":     "getHeader",
-		"slot":       slot,
-		"parentHash": parentHashHex,
-		"pubkey":     pubkey,
-	})
-	log.Debug("getHeader")
 
-	_slot, err := strconv.ParseUint(slot, 10, 64)
+	publicKey, err := types.HexToPubkey(vars["pubkey"])
 	if err != nil {
-		m.respondError(w, http.StatusBadRequest, errInvalidSlot.Error())
+		m.respondError(w, http.StatusBadRequest, errInvalidPubkey.Error())
 		return
 	}
 
-	if len(pubkey) != 98 {
-		m.respondError(w, http.StatusBadRequest, errInvalidPubkey.Error())
+	slot, err := strconv.ParseUint(vars["slot"], 10, 64)
+	if err != nil {
+		m.respondError(w, http.StatusBadRequest, errInvalidSlot.Error())
 		return
 	}
 
-	if len(parentHashHex) != 66 {
+	parentHash := types.Hash{}
+	if err := parentHash.UnmarshalText([]byte(vars["parent_hash"])); err != nil {
 		m.respondError(w, http.StatusBadRequest, errInvalidHash.Error())
 		return
 	}
 
+	log := m.log.WithFields(logrus.Fields{
+		"method":     "getHeader",
+		"slot":       slot,
+		"parentHash": parentHash.String(),
+		"pubkey":     publicKey.String(),
+	})
+	log.Debug("getHeader")
+
 	var mu sync.Mutex
 	relays := make(map[string][]string) // relays per blockHash
 	result := bidResp{}
 
 	ua := UserAgent(req.Header.Get("User-Agent"))
 
+	result.proposerPublicKeys = append(result.proposerPublicKeys, publicKey)
```

Comment:
shouldn't there be only a single proposer per slot?

Reply:
Yep it was a misunderstood on my side. Fixed in eb6cbc1543f2f572b298ec6c663f9c34263d7793.